### ISSUE TITLE
TESTING OF ALPHAGEOMETRY

### ISSUE THREAD

**Issue by Ehisnet** [google-deepmind/alphageometry#60]
Have anyone been able to successfully run all the tests for alphageometry successfully. please if you have can you tell us what you did to pass all the tests.
Also, have anyone been able to test other geometry olympiad problems aside the ones in the provided in the package. If yes how were you able to translate the problem into the format used by alphageonmetry without having syntax errors.

**Comment by unhandyandy** [comment]
Yeah, it would be nice to have guide to the language used to write problems.

I got it running, I'll try the test suite next time I have a chance.

**Comment by Ehisnet** [comment]
In which environment were you able to run it

**Comment by unhandyandy** [comment]
On the bigred200 HPC at IU.  It successfully solved the first example problem.  I used python3 v. 10.

**Comment by Ehisnet** [comment]
I used PuTTy and Xming alongside Linux WSL Server in my local machine

**Comment by unhandyandy** [comment]
I ran the tests, and all passed except the last one, lm_inference_test.py, which failed with 
`ModuleNotFoundError: No module named 'jax'`
which seems strange, because the jax library is right there with the others, in
`lib/python3.10/site-packages/`

I don't know enough about python to diagnose the problem.  Isn't jax necessary to solve problems?  And it does solve the orthocenter problem.

I'll try solving the other example problems...

**Comment by Ehisnet** [comment]
The last test failing is a general issue observed by all including me.

**Comment by unhandyandy** [comment]
Strange, because it evidently loads OK when alphageometry.py is run in solving an example problem.  What's the difference?

**Comment by Ehisnet** [comment]
Difference in outputs.

**Comment by iub-organizing-collective** [comment]
What do you mean?

**Comment by unhandyandy** [comment]
OK, but I'm not getting an assertion error, I'm getting ModuleNotFoundError.

Also, it's actually the penultimate test that fails,  alphageometry_test.py.  Apparently that's the first one that loads jax?

**Comment by Ehisnet** [comment]
Did you try to install the missing module and retesting

**Comment by iub-organizing-collective** [comment]
As I said, it was never missing - AG was able to solve problems fine.
The problem was that I wasn't running run_tests.sh inside the virtualenv.  Have you tried that?

**Comment by unhandyandy** [comment]
Sorry for the confusion: I'm also iub-organizing-collective.

**Comment by unhandyandy** [comment]
Solved the first imo problem.  

The only test the failed was #14 , which apparently is not important.

**Comment by Ehisnet** [comment]
try if you can solve all the 30 problems

**Comment by unhandyandy** [comment]
Which one are people getting stuck on?

**Comment by unhandyandy** [comment]
It failed to solve the second problem, but did not produce any errors.  That doesn't indicate a problem with the installation.

**Comment by unhandyandy** [comment]
Increasing the values of beam width and depth is probably necessary for some problems.

**Comment by unhandyandy** [comment]
@Ehisnet I'm not sure what you mean by "scattered".  Unfortunately AG will often relabel the points - is that what you mean?

What problem was this?

**Comment by Ehisnet** [comment]
check my screenshots you will understand what i meant

**Comment by unhandyandy** [comment]
I see the screenshot, but I'm not sure what you mean.  Ag also randomizes points, so its diagram may not look similar to the one you draw yourself.

**Comment by Ehisnet** [comment]
> @Ehisnet I'm not sure what you mean by "scattered". Unfortunately AG will often relabel the points - is that what you mean?
> 
> What problem was this?

@unhandyandy  i mean the letters sometimes are not usually rightly positioned in the diagram

**Comment by unhandyandy** [comment]
@Ehisnet Ah yes, sometimes it's hard to tell which label goes with which point.  That just seems to be an inevitable problem with algorithm for placing labels automatically in an image.

**Comment by tpgh24** [comment]
AlphaGeometry is sensitive to OS and Python version. I made some improvements in a fork repository and have some ideas to further improve the performance of AlphaGeometry, check out [AG4Masses](https://github.com/tpgh24/ag4masses) and [issue 110](https://github.com/google-deepmind/alphageometry/issues/110). I also shared my experiences with AG, including a detailed explanation of the problem definition language.

**Comment by leif-pixel** [comment]
How did all of you run the program? I am a beginner and just cloned https://github.com/google-deepmind/alphageometry.git, but I don't know how to continue. After cloning it, I tried the instructions but it also did not work. Any help? Thanks!
